README.md

```diff
@@ -1,21 +1,68 @@
-# NervesSsh
+# NervesSSH
+[![CircleCI](https://circleci.com/gh/nerves-project/nerves_ssh.svg?style=svg)](https://circleci.com/gh/nerves-project/nerves_ssh)
+[![Hex version](https://img.shields.io/hexpm/v/nerves_ssh.svg "Hex version")](https://hex.pm/packages/nerves_ssh)
 
-**TODO: Add description**
+Manage a SSH daemon and subsystems on Nerves devices.
 
 ## Installation
 
-If [available in Hex](https://hex.pm/docs/publish), the package can be installed
-by adding `nerves_ssh` to your list of dependencies in `mix.exs`:
+To use, add the dependency to your `mix.exs`:
 
 ```elixir
 def deps do
   [
-    {:nerves_ssh, "~> 0.1.0"}
+    {:nerves_ssh, "~> 0.1.0", targets: @all_targets}
   ]
 end
 ```
 
-Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
-and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
-be found at [https://hexdocs.pm/nerves_ssh](https://hexdocs.pm/nerves_ssh).
+## Goals
+
+- [X] Support public key authentication
+- [ ] Support username/password authentication
+- [ ] Device generated Server certificate and key
+
+## Usage
+
+<!-- MDOC !-->
+
+A wrapper around SSH daemon focused purely on the context of Nerves devices.
+
+It keeps `sshd` under supervision and supports managing it at runtime so it
+can be enabled/disabled on the fly. This is most useful in cases where you
+want SSH access disabled by default, but want the ability to quickly enable
+for debugging purposes, such as starting SSH on major application failures.
+
+While this will be started with your application, it is generally a good
+idea to separate connection level processes from your main application via
+`shoehorn`. In the event of your application failure, that still keeps the
+ability to conenct to the device over SSH:
+
+```elixir
+config :shoehorn,
+  init: [:nerves_runtime, :vintage_net, :nerves_ssh]
+```
+
+NervesSSH also starts a daemon by default. If you wish to delay this and
+start SSH daemon yourself at runtime, you can specify the `auto_start: false`
+option in your application config:
+
+```elixir
+config :nerves_ssh, auto_start: false
+```
+
+You'll then need to call `NervesSSH.enable/1` later in your application
+when needing to start the SSH daemon
+
+## Configuration
+
+NervesSSH supports a few pieces of configuration via the application config
+or when calling `NervesSSH.enable/1`:
+
+* `auto_start` - boolean dictating if SSH daemon start on applicaiton startup
```

Comment:
Ha! All the way to the dictionary with this.

This makes sense. Maybe `enabled` is a better option. I was already debating changing the API to `start` and `stop` functions as well

Does `enabled` make sense for run the OTP app, but not the underlying wrapper process? Maybe `run_on_startup`?

Going to sleep on it a bit and see if I think of anything else

Reply:
I've been thinking about a `run_on_startup` scenario:

1. Set `run_on_startup` to false in `config.exs`
2. Boot and start all OTP applications like normal. `nerves_ssh`'s OTP application starts, but the ssh daemon doesn't.
3. Call whatever API starts the ssh daemon
4. Through some mechanism, the `nerves_ssh` application stops.
5. Does `shoehorn` restart it or reboot? Let's say the the answer is to restart. Is the ssh daemon started?

Also, and maybe more importantly, what are the use cases for not starting the ssh daemon? The one that I can think of is that ssh is a feature that can be enabled or disabled by the user at runtime. Are there others?

Reply:
Another case is the need to restrict open ports by default, like in a production system, but still want the ability to easily enable SSH when the need arises (debugging, sftp a file, etc)

In that case `run_on_startup` is not the right convention either because the setting would more of an "off by default" scenario and always require explicitly starting SSH. I.e. if `nerves_ssh` application stops, I wouldn't the SSH daemon started again when the application comes until I call the API